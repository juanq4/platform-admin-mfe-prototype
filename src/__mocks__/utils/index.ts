export * from "./helpers";
export * from "./permissions";
export * from "./wrappers";
