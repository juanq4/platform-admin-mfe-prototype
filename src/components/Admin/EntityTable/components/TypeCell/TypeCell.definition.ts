import { AutomationListId, IdModelBase } from "@q4/nimbus-ui";
import type { OrganizationType } from "@q4/platform-definitions";
import type { Key } from "react";
import type { EntityTableCellPropsBase } from "../../EntityTable.definition";

export type TypeCellProps = EntityTableCellPropsBase<OrganizationType>;

export enum TypeCellLabel {
  Corporate = "Corporate",
  Agency = "Agency",
  Admin = "Admin",
}

export class TypeCellListId extends AutomationListId<TypeCellIdModel> {
  constructor(parentId: string, prefix = "TypeCell-", postfix?: string) {
    super(parentId, prefix, TypeCellIdModel, postfix);
  }
}

export class TypeCellIdModel extends IdModelBase {
  constructor(id: IdModelBase["id"], index?: Key, postfix?: string) {
    super(id, index, postfix);
  }
}
