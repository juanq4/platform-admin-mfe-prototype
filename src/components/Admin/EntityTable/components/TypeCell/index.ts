export * from "./TypeCell.component";
export * from "./TypeCell.definition";
export * from "./TypeCell.utils";
