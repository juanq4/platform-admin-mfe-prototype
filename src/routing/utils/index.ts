export * from "./api";
export * from "./s3checks";
export * from "./versionManager";
