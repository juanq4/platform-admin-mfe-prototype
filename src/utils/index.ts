export * from "./api";
export * from "./entitlement";
export * from "./error";
export * from "./organization";
export * from "./permission";
export * from "./role";
export * from "./route";
export * from "./users";
