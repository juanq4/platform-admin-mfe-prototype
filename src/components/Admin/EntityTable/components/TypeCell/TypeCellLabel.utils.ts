import { OrganizationType } from "@q4/platform-definitions";
import { TypeCellLabel } from "./TypeCell.definition";

export function getTypeCellLabelFromOrganizationType(type: OrganizationType | null): TypeCellLabel {
  switch (type) {
    case OrganizationType.AGENCY:
      return TypeCellLabel.Agency;
    case OrganizationType.ADMIN:
      return TypeCellLabel.Admin;
    case OrganizationType.CORP:
    default:
      return TypeCellLabel.Corporate;
  }
}
