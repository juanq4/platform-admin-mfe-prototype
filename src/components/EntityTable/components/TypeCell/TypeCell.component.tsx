import { memo, useMemo } from "react";
import { EntityTableClassName } from "../../EntityTable.definition";
import { TypeCellIdModel } from "./TypeCell.definition";
import type { TypeCellProps } from "./TypeCell.definition";
import { getTypeCellLabelFromOrganizationType } from "./TypeCellLabel.utils";

const TypeCellBase = (props: TypeCellProps): JSX.Element => {
  const { id, value } = props;
  const idModel = useMemo(() => new TypeCellIdModel(id), [id]);
  const content = useMemo(() => {
    const type = getTypeCellLabelFromOrganizationType(value);

    return <span>{type}</span>;
  }, [value]);

  return (
    <div id={idModel.id} className={EntityTableClassName.TableCell}>
      {content}
    </div>
  );
};

export const TypeCell = memo(TypeCellBase);
