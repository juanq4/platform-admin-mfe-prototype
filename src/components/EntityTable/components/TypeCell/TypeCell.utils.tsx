import { TypeCell } from "./TypeCell.component";
import type { TypeCellListId, TypeCellProps } from "./TypeCell.definition";

export function mapIdToTypeCell(params: TypeCellProps, listId: TypeCellListId): JSX.Element {
  const id = listId?.getId(params?.data?.id);

  return <TypeCell {...params} id={id} />;
}
