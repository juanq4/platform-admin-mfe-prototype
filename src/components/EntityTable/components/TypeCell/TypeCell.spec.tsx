import { OrganizationType } from "@q4/platform-definitions";
import React from "react";
import { render, screen } from "../../../../utils/testUtils";
import { TypeCellIdModel, TypeCellListId } from "../TypeCell/TypeCell.definition";
// import { getAppWrapper } from "../../../../__mocks__";
import { TypeCell } from "./TypeCell.component";
import { mapIdToTypeCell } from "./TypeCell.utils";

const mockTypeCellValues = [null, OrganizationType.AGENCY, OrganizationType.CORP, OrganizationType.ADMIN];

function testTypeCell(id: string): void {
  const element = screen.queryByTestId(id);
  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
}

describe("TypeCell Component", () => {
  const idModel = new TypeCellIdModel("TypeCell");

  test("7386714: [Given] no props are provided [Expect] the component should render", () => {
    render(<TypeCell value={null} />, { container: getAppWrapper() });
    const rootElement = screen.queryByTestId("root");
    expect(rootElement).toMatchSnapshot();
  });

  test("4754437: [Given] I'm a user of the organizations admin view[When] the organization has a type of ‘Agency’ [Then] I can see the word ‘Agency’ listed under the “Type” column", () => {
    render(<TypeCell id={idModel.id} value={OrganizationType.AGENCY} />);
    testTypeCell(idModel.id);
  });

  test("4754438: [Given] I'm a user of the organizations admin view [When] the organization has a type of “Corporate”[Then] I can see the word “Corporate” listed under the “Type” column", () => {
    render(<TypeCell id={idModel.id} value={OrganizationType.CORP} />);
    testTypeCell(idModel.id);
  });

  test("4754439: [Given] I'm a user of the organizations admin view [When] the organization in that row does not have a type [Then] I can see the word “Corporate” listed under the “Type” column", () => {
    render(<TypeCell id={idModel.id} value={null} />);
    testTypeCell(idModel.id);
  });
});

describe("mapIdToTypeCell", () => {
  const listId = new TypeCellListId("MockCell");
  const mockData = { id: "4dec165f-4093-4182-bd72-88754" };

  test("7386715: [Given] the function is passed not no props [Expect] the component should render", () => {
    render(mapIdToTypeCell(null, null), { container: getAppWrapper() });
    const rootElement = screen.queryByTestId("root");
    expect(rootElement).toMatchSnapshot();
  });

  test("7434028: [Given] value is 'null' [Expect] the status to be 'Corporate'", () => {
    render(mapIdToTypeCell({ data: { ...mockData }, value: mockTypeCellValues[0] }, listId));
    const id = listId.getId(mockData.id);
    testTypeCell(id);
  });

  test("7434029: [Given] value is 'agency' [Expect] the status to be 'Agency'", () => {
    render(mapIdToTypeCell({ data: { ...mockData }, value: mockTypeCellValues[1] }, listId));
    const id = listId.getId(mockData.id);
    testTypeCell(id);
  });

  test("7434030: [Given] value is 'corporate' [Expect] the status to be 'Corporate'", () => {
    render(mapIdToTypeCell({ data: { ...mockData }, value: mockTypeCellValues[2] }, listId));
    const id = listId.getId(mockData.id);
    testTypeCell(id);
  });
});
