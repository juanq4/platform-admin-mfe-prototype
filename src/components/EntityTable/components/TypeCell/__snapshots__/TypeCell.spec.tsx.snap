// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeCell Component 4754437: [Given] I'm a user of the organizations admin view[When] the organization has a type of ‘Agency’ [Then] I can see the word ‘Agency’ listed under the “Type” column 1`] = `
<div
  class="entity-table_table-cell"
  id="TypeCell"
>
  <span>
    Agency
  </span>
</div>
`;

exports[`TypeCell Component 4754438: [Given] I'm a user of the organizations admin view [When] the organization has a type of “Corporate”[Then] I can see the word “Corporate” listed under the “Type” column 1`] = `
<div
  class="entity-table_table-cell"
  id="TypeCell"
>
  <span>
    Corporate
  </span>
</div>
`;

exports[`TypeCell Component 4754439: [Given] I'm a user of the organizations admin view [When] the organization in that row does not have a type [Then] I can see the word “Corporate” listed under the “Type” column 1`] = `
<div
  class="entity-table_table-cell"
  id="TypeCell"
>
  <span>
    Corporate
  </span>
</div>
`;

exports[`TypeCell Component 7386714: [Given] no props are provided [Expect] the component should render 1`] = `
<div
  id="root"
>
  <div
    class="entity-table_table-cell"
  >
    <span>
      Corporate
    </span>
  </div>
</div>
`;

exports[`mapIdToTypeCell 7386715: [Given] the function is passed not no props [Expect] the component should render 1`] = `
<div
  id="root"
>
  <div
    class="entity-table_table-cell"
  >
    <span>
      Corporate
    </span>
  </div>
</div>
`;

exports[`mapIdToTypeCell 7434028: [Given] value is 'null' [Expect] the status to be 'Corporate' 1`] = `
<div
  class="entity-table_table-cell"
  id="MockCellTypeCell-4dec165f-4093-4182-bd72-88754"
>
  <span>
    Corporate
  </span>
</div>
`;

exports[`mapIdToTypeCell 7434029: [Given] value is 'agency' [Expect] the status to be 'Agency' 1`] = `
<div
  class="entity-table_table-cell"
  id="MockCellTypeCell-4dec165f-4093-4182-bd72-88754"
>
  <span>
    Agency
  </span>
</div>
`;

exports[`mapIdToTypeCell 7434030: [Given] value is 'corporate' [Expect] the status to be 'Corporate' 1`] = `
<div
  class="entity-table_table-cell"
  id="MockCellTypeCell-4dec165f-4093-4182-bd72-88754"
>
  <span>
    Corporate
  </span>
</div>
`;
